### DISCUSSION TITLE
How to easily and elegantly roll-up nats-server instances to the latest version within a cluster

### DISCUSSION THREAD

**Discussion by Doslin** [nats-io/nats-server#3778] [Q&A]
How to easily and elegantly roll-up nats-server instances to the latest version within a cluster

**Comment by Doslin** [comment]
> How to easily and elegantly roll-up nats-server instances to the latest version within a cluster

The restart command `nats-server --signal reload=1` doesn't seem to work

**Comment by derekcollison** [comment]
Shutdown with lameduck signal.
Start new version server, wait for /healthz to return ok.
Move to next server.

**Reply by bruth** [reply]
@Doslin For the lameduck signal, you can do `nats-server --signal ldm=1` (assuming PID is 1) or send a `SIGUSR2` signal to the process.

**Reply by bruth** [reply]
For the health check, once the new server starts, you can do something like: `curl --fail --silent --retry 5 --retry-delay 1 http://my-nats-server:8222/healthz > /dev/null` to assert it is healthy. This translates to retry this endpoint 5 times until it returns a 200, otherwise considered fail with a non-zero exit code.

**Comment by derekcollison** [comment]
This is built in to the helm operator.
